Classes/HK417Fire.uc

```diff
@@ -1,5 +1,26 @@
 class HK417Fire extends KFFire;
 
+//adds faked recoil to 3d scope zoom
+event ModeDoFire()
+{
+    if( KFWeapon(Weapon).bAimingRifle )
+    {
+        if ( KFWeapon(Weapon).KFScopeDetail != KF_TextureScope)
+        {
+            ShakeOffsetMag.X=2.0; //faked recoil
```

Comment:
Put it in a variable instead of hard-coding. `ScopedShakeOffsetMag` and `ScopedShakeOffsetRate`.